crates/node-builder/src/components/exex.rs

```diff
@@ -0,0 +1,106 @@
+use std::{
+    collections::VecDeque,
+    future::Future,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+use reth_primitives::BlockNumber;
+use reth_provider::{CanonStateNotification, CanonStateNotifications};
+use tokio::sync::{mpsc::UnboundedSender, watch};
+use tokio_util::sync::PollSender;
+
+// todo
+// - exex height
+// - exex channel size
+// - manager buffer metrics
+// - throughput
+struct ExExMetrics;
+
+/// The execution extension manager.
+///
+/// The manager is responsible for:
+///
+/// - Receiving relevant events from the rest of the node, and sending these to the execution
+/// extensions
+/// - Backpressure
+/// - Error handling
+/// - Monitoring
+///
+/// TBD
+struct ExExManager {
+    // todo: exex receiver - get exex events
+    /// Channels from the manager to execution extensions.
+    exex_tx: Vec<PollSender<CanonStateNotification>>,
```

Comment:
this needs to be it's own type because we need to monitor which event we're currently sending to this exex